contracts/external/gmxV2/GmxV2IsolationModeTokenVaultV1.sol

```diff
@@ -0,0 +1,421 @@
+        _setSourceIsWrapper(_sourceIsWrapper);
+    }
+
+    function setShouldSkipTransfer(
+        bool _shouldSkipTransfer
+    )
+    external
+    onlyVaultFactory(msg.sender) {
+        _setShouldSkipTransfer(_shouldSkipTransfer);
+    }
+
+    // @audit Does this need to be requireNotFrozen? I don't think so but want to confirm
+    function executeDepositIntoVault(
+        address _from,
+        uint256 _amount
+    )
+    public
+    override
+    onlyVaultFactory(msg.sender) {
+        _setVirtualBalance(getVirtualBalance() + _amount);
+
+        if (!isShouldSkipTransfer()) {
+            if (!isSourceIsWrapper()) {
+                IERC20(UNDERLYING_TOKEN()).safeTransferFrom(_from, address(this), _amount);
+            }
+            else {
+                IERC20(UNDERLYING_TOKEN()).safeTransferFrom(
+                    address(registry().gmxV2WrapperTrader()),
+                    address(this),
+                    _amount
+                );
+                _setSourceIsWrapper(false);
+            }
+            _compareVirtualToRealBalance();
+        } else {
+            // @todo confirm the vault is frozen
+            _setShouldSkipTransfer(false);
+        }
+    }
+
+    // @audit Does this need to be requireNotFrozen? I don't think so but want to confirm
+    function executeWithdrawalFromVault(
+        address _recipient,
+        uint256 _amount
+    )
+    public
+    override
+    onlyVaultFactory(msg.sender) {
+        _setVirtualBalance(getVirtualBalance() - _amount);
+
+        if (!isShouldSkipTransfer()) {
+                IERC20(UNDERLYING_TOKEN()).safeTransfer(_recipient, _amount);
+                _compareVirtualToRealBalance();
+        }
+        else {
+            _setShouldSkipTransfer(false);
+        }
+    }
+
+    function getVirtualBalance() public view returns (uint256) {
+        return _getUint256(_VIRTUAL_BALANCE_SLOT);
+    }
+
+    function isVaultFrozen() public view returns (bool) {
+        return _getUint256(_VAULT_FROZEN_SLOT) == 1;
+    }
+
+    function isSourceIsWrapper() public view returns (bool) {
+        return _getUint256(_SOURCE_IS_WRAPPER_SLOT) == 1;
+    }
+
+    function isShouldSkipTransfer() public view returns (bool) {
+        return _getUint256(_SHOULD_SKIP_TRANSFER_SLOT) == 1;
+    }
+
+    function registry() public view returns (IGmxRegistryV2) {
+        return IGmxV2IsolationModeVaultFactory(VAULT_FACTORY()).gmxRegistryV2();
+    }
+
+    function dolomiteRegistry()
+        public
+        override
+        view
+        returns (IDolomiteRegistry)
+    {
+        return registry().dolomiteRegistry();
+    }
+
+    // ==================================================================
+    // ======================== Internal Functions ========================
+    // ==================================================================
+
+    // @follow-up If these internal functions need to be used when frozen, we can overwrite external ones instead
```

Comment:
Haven't thought about this yet. Still am on the first pass through of code review but it's a good thought.